applications/serial_lte_modem/src/mqtt_c/slm_at_mqtt.c

```diff
@@ -459,6 +473,55 @@ static int do_mqtt_subscribe(uint16_t op,
 	return err;
 }
 
+/* Handles AT#XMQTTCFG commands. */
+int handle_at_mqtt_config(enum at_cmd_type cmd_type)
+{
+	int err = -EINVAL;
+	uint16_t keep_alive = CONFIG_MQTT_KEEPALIVE;
+	uint16_t clean_session = CONFIG_MQTT_CLEAN_SESSION;
+	uint32_t param_count = at_params_valid_count_get(&slm_at_param_list);
+
+	switch (cmd_type) {
+	case AT_CMD_TYPE_SET_COMMAND:
+		size_t clientid_sz = sizeof(mqtt_clientid);
+
+		err = util_string_get(&slm_at_param_list, 1, mqtt_clientid, &clientid_sz);
+		if (err) {
+			return err;
+		}
+		if (param_count > 2) {
+			err = at_params_unsigned_short_get(&slm_at_param_list, 2, &keep_alive);
+			if (err) {
+				return err;
+			}
+		}
+		if (param_count > 3) {
+			err = at_params_unsigned_short_get(&slm_at_param_list, 3, &clean_session);
+			if (err) {
+				return err;
+			}
+		}
+		err = do_mqtt_config(keep_alive, (uint8_t)clean_session);
+		break;
+
+	case AT_CMD_TYPE_READ_COMMAND:
+		rsp_send("\r\n#XMQTTCFG: \"%s\",%d,%d\r\n",
+			 mqtt_clientid, client.keepalive, client.clean_session);
+		err = 0;
+		break;
+
+	case AT_CMD_TYPE_TEST_COMMAND:
+		rsp_send("\r\n#XMQTTCFG: <client_id>,<keep_alive>,<clean_session>\r\n");
+		err = 0;
+		break;
+
+	default:
+		break;
+	}
+
+	return err;
+}
+
 /* Handles AT#XMQTTCON commands. */
 int handle_at_mqtt_connect(enum at_cmd_type cmd_type)
```

Comment:
What was the reasoning for not having the parameters in the MQTTCON?

Reply:
?

Reply:
Keepalive and clean session parameters could also have been added to MQTTCON, and be in effect for a particular connection. Instead they are set in separate AT-command and are in effect until the device is reset.

Reply:
In my opinion #XMQTTCON should only ask for parameter as in MQTT URI scheme.
But if you think that is better, I can close this PR for a new one.